### PR TITLE
document direct ts executing adapters

### LINKED ISSUE #1236: Feature idea: execute TypeScript adapters directly with ts-node
The basic idea is that we can invoke TypeScript adapters with https://github.com/TypeStrong/ts-node, a TypeScript wrapper around node. We could detect if the `main` file has a `.ts` extension and invoke it the following way instead:

```
node -r ts-node/register/transpile-only main.ts ...arguments...
```

Questions left to be solved (would be best if these are not necessary):
- [ ] Does the adapter need to include `tsconfig.json`?
- [ ] Does the adapter need to include `@types` packages as `devDependencies`?

### PR DESCRIPTION
**Pull request by foxriver76**
- closes #1236

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,6 +114,20 @@ If needed, especially for low memory situations the memory limit for all adapter
 }
 ```
 
+### Directly exeuting Typescript adapters
```
Comment (AlCalzone) [inline README.md:117]:
```suggestion
### Directly executing TypeScript adapters
```

File: README.md [hunk 2]
```diff
@@ -114,6 +114,20 @@ If needed, especially for low memory situations the memory limit for all adapter
...
+**Feature status:** Technology preview since js-controller 3.3.0
+
+The js-controller is able to execute `.ts` files, which removes the need of compiling to Javascript first.
```
Comment (AlCalzone) [inline README.md:120]:
```suggestion
The js-controller is able to execute `.ts` files, which removes the need of compiling to JavaScript first.
```

File: README.md [hunk 3]
```diff
@@ -114,6 +114,20 @@ If needed, especially for low memory situations the memory limit for all adapter
...
+To use this feature, simply define the adapter main file as the required `.ts` file.
```
Comment (AlCalzone) [inline README.md:121]: Additional points to consider:
- The `src` directory needs to be released as part of the npm package (--> .npmignore / `package.json files`)
- The `build` directory probably shouldn't be created (disable compilation scripts/automations) or at least not checked in to GitHub / released to `npm`.
- The adapter is compiled at every startup (with the very performant ESBuild, but that still takes some additional resources)

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (foxriver76) [reply README.md:120]: :+1: then also TypeScript

Comment (foxriver76) [reply README.md:120]: ah yes other comment

File: README.md (hunk 3 above)
Comment (foxriver76) [reply README.md:121]: Quesiton is do we want to document this here?

Comment (AlCalzone) [reply README.md:121]: Maybe only the last point as a thing to consider. The others are more suited for the dev docs.

**Comment by Apollon77** [comment]
LGTM

**Pull request merged**